### PR TITLE
Single robot replacement

### PR DESCRIPTION
**Pull request by SimLemay**

### CODE REVIEW

File: Controller/MainController.py [hunk 1]
```diff
@@ -462,4 +462,5 @@ def save_teams_formation(self):
         self.teams_formation = self.view_field_screen.get_teams_formation()
 
     def restore_teams_formation(self):
-        self.grsim_sender.set_robot_positions(self.teams_formation)
+        self.grsim_sender.set_robots_position(self.teams_formation)
```
Comment (TwistedSim) [inline Controller/MainController.py]: property! :)

**Comment by SimLemay** [comment]
Sorry for the last 2 commits

File: View/FieldView.py [hunk 2]
```diff
@@ -282,6 +285,7 @@ def select_robot(self, bot_id, team_color):
         for mob in mobs:
             if mob.get_id() == bot_id:
                 mob.select()
+                self.selected_mob = mob
             else:
                 mob.deselect()
```
Comment (TwistedSim) [inline View/FieldView.py]:
```python
map(lambda mob: mob.deselect(), mobs)
mob = filter(lambda mob: mob.get_id() == bot_id, mobs)[0]
mob.select()
```
or
```python
map(lambda mob: mob.deselect(), mobs)
mob = next(mob for mob in mobs if mob.get_id() == bot_id)
mob.select()
```

deselect should be handle by the graph, not the user.
also get_id shoul be an attribute. no need for a get in python.

File: View/FieldView.py [hunk 3]
```diff
@@ -366,8 +370,14 @@ def mouseDoubleClickEvent(self, event):
             x, y = QtToolBox.field_ctrl.convert_screen_to_real_pst(event.pos().x(), event.pos().y())
 
             if event.buttons() == Qt.RightButton:
-                self.controller.grsim_sender.set_ball_position((x, y))
-                # If we are playing with tactics we handle double left click
+                if QApplication.keyboardModifiers() == Qt.ControlModifier and self.selected_mob is not None:
+                    direction = self.selected_mob.get_position_in_real()[2]
```
Comment (TwistedSim) [inline View/FieldView.py]: should be a property

File: View/FieldView.py [hunk 4]
```diff
@@ -366,8 +370,14 @@ def mouseDoubleClickEvent(self, event):
...
+                    team_color_is_yellow = "yellow" == self.controller.get_team_color()
+                    self.controller.grsim_sender.set_robot_position(x, y, direction, self.selected_mob.get_id(),
+                                                                    team_color_is_yellow)
+                else:
+                    self.controller.grsim_sender.set_ball_position((x, y))
```
Comment (TwistedSim) [inline View/FieldView.py]: same here

File: main.py [hunk 5]
```diff
@@ -24,7 +24,7 @@ def load_config(path):
     config_parser = ConfigParser(allow_no_value=False)
     try:
         print("Loading", path, " port configuration file.")
-        config_parser.read_file(open(path))
+        config_parser   .read_file(open(path))
```
Comment (TwistedSim) [inline main.py]: this is strange, why spaces?

File: Controller/MainController.py (hunk 1 above)
Comment (PhiBabin) [reply Controller/MainController.py]: Dans ui-debug c'est correcte puique le set/get règne en roi

**Review by PhiBabin — APPROVED** [review]

File: Controller/MainController.py (hunk 1 above)
Comment (TwistedSim) [reply Controller/MainController.py]: pas une excuse valable :P.

**Review by PhiBabin — APPROVED** [review]

**Pull request merged**